sentry-android-ndk/src/main/jni/sentry.c

```diff
@@ -11,6 +11,90 @@ struct transport_options {
 
 struct transport_options g_transport_options;
 
+JNIEXPORT void JNICALL Java_io_sentry_android_ndk_NdkScopeObserver_nativeSetTag(JNIEnv *env, jclass cls, jstring key, jstring value) {
+    if (key == NULL) {
+        return;
+    }
+    const char *charKey = (*env)->GetStringUTFChars(env, key, 0);
+    const char *charValue = NULL;
+    if (value == NULL) {
```

Comment:
how does this work? you call `GetStringUTFChars` on a `NULL`?

I would just guard with `if (!key || !value) { return; }` on top…

Reply:
It's a bug